contracts/schemes/Competition.sol

```diff
@@ -0,0 +1,349 @@
+        proposals[proposalId].contributionRewardExt = _contributionRewardExt;
+        proposals[proposalId].reputationReward = uint256(_reputationChange);
+        proposals[proposalId].nativeTokenReward = _rewards[0];
+        proposals[proposalId].ethReward = _rewards[1];
+        proposals[proposalId].externalTokenReward = _rewards[2];
+
+        emit NewCompetitionProposal(
+            proposalId,
+            numberOfWinners,
+            proposals[proposalId].rewardSplit,
+            startTime,
+            proposals[proposalId].votingStartTime,
+            proposals[proposalId].endTime,
+            proposals[proposalId].numberOfVotesPerVoters,
+            _contributionRewardExt
+        );
+    }
+
+    /**
+    * @dev suggest a competion suggestion
+    * @param _proposalId the proposal id this suggestion is referring to.
+    * @param _descriptionHash a descriptionHash of the suggestion.
+    * @return suggestionId the suggestionId.
+    */
+    function suggest(
+            bytes32 _proposalId,
+            string calldata _descriptionHash
+    )
+    external
+    returns(uint256)
+    {
+      // solhint-disable-next-line not-rely-on-time
+        require(proposals[_proposalId].startTime <= now, "competition not started yet");
+        // solhint-disable-next-line not-rely-on-time
+        require(proposals[_proposalId].endTime > now, "competition ended");
+        suggestionsCounter = suggestionsCounter.add(1);
+        suggestions[suggestionsCounter].proposalId = _proposalId;
+        suggestions[suggestionsCounter].suggester = msg.sender;
+        emit NewSuggestion(_proposalId, suggestionsCounter, _descriptionHash, msg.sender);
+        return suggestionsCounter;
+    }
+
+    /**
+    * @dev vote on a suggestion
+    * @param _suggestionId suggestionId
+    * @return bool
+    */
+    function vote(uint256 _suggestionId)
+    external
+    returns(bool)
+    {
+        bytes32 proposalId = suggestions[_suggestionId].proposalId;
+        require(proposalId != bytes32(0), "suggestion not exist");
+        setSnapshotBlock(proposalId);
+        Proposal storage proposal = proposals[proposalId];
+        // solhint-disable-next-line not-rely-on-time
+        require(proposal.endTime > now, "competition ended");
+        Suggestion storage suggestion = suggestions[_suggestionId];
+        require(suggestion.votes[msg.sender] == 0, "already voted on this suggestion");
+        require(proposal.votesPerVoter[msg.sender] < proposal.numberOfVotesPerVoters,
+        "exceed number of votes allowed");
+        proposal.votesPerVoter[msg.sender] = proposal.votesPerVoter[msg.sender].add(1);
+        suggestion.votes[msg.sender] = suggestion.votes[msg.sender].add(1);
+        Avatar avatar = ContributionRewardExt(proposal.contributionRewardExt).avatar();
+        uint256 reputation = avatar.nativeReputation().balanceOfAt(msg.sender, proposals[proposalId].snapshotBlock);
+        require(reputation > 0, "voter has no reputation");
```

Comment:
This is a very confusing error message if you do have reputation, but did not have it in the snapshot. Change to` voter had no reputation when ..` (??)